Apache Pekko doesn't support autogenerated TLS certs

The Play 3.0 documentation [states](https://www.playframework.com/documentation/3.0.2/ConfiguringHttps#SSL-Certificates-from-a-keystore):

> By default, Play will generate itself a self-signed certificate ...

But after upgradinig to Play 3.0 and thereby switching from Akka to Apache Pekko, when ich run

```bash
<my-project-executable> -Dhttp.port=disabled -Dhttps.port=9443 -Dpidfile.path=/dev/null -Dlogger.resource=dev-logback.xml
```

I get the following error message

`[error] play.core.server.PekkoHttpServer - Cannot load SSL context`

Full stack trace below:  
<details>
<code>
java.lang.reflect.InvocationTargetException: null
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.>
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAcces>
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at play.core.server.ssl.ServerSSLEngine$.createScalaSSLEngineProvider(ServerSSLEngine.scala:122)
        at play.core.server.ssl.ServerSSLEngine$.createSSLEngineProvider(ServerSSLEngine.scala:39)
        at play.core.server.PekkoHttpServer.sslContext$lzyINIT1(PekkoHttpServer.scala:245)
        at play.core.server.PekkoHttpServer.sslContext(PekkoHttpServer.scala:245)
        at play.core.server.PekkoHttpServer.$init$$$anonfun$5(PekkoHttpServer.scala:260)
Caused by: java.lang.IllegalAccessError: class com.typesafe.sslconfig.ssl.FakeKeyStore (in unnamed module @0xfdc81>
        at com.typesafe.sslconfig.ssl.FakeKeyStore.certificateTooWeak(FakeKeyStore.scala:169)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.shouldGenerate$$anonfun$2$$anonfun$1(FakeKeyStore.scala:152)
        at scala.Option.exists(Option.scala:406)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.shouldGenerate$$anonfun$1(FakeKeyStore.scala:152)
        at scala.collection.IterableOnceOps.exists(IterableOnce.scala:604)
        at scala.collection.IterableOnceOps.exists$(IterableOnce.scala:601)
        at scala.collection.AbstractIterator.exists(Iterator.scala:1300)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.shouldGenerate(FakeKeyStore.scala:152)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.createKeyStore(FakeKeyStore.scala:179)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.keyManagerFactory(FakeKeyStore.scala:218)
[debug] org.apache.pekko.io.TcpListener - Successfully bound to /[0:0:0:0:0:0:0:0]:9443
[info] play.core.server.PekkoHttpServer - Listening for HTTPS on /[0:0:0:0:0:0:0:0]:9443
[warn] play.core.server.ssl.DefaultSSLEngineProvider - Using generated key with self signed certificate for HTTPS.>
Oops, cannot start the server.
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.>
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAcces>
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at play.core.server.ssl.ServerSSLEngine$.createScalaSSLEngineProvider(ServerSSLEngine.scala:122)
        at play.core.server.ssl.ServerSSLEngine$.createSSLEngineProvider(ServerSSLEngine.scala:39)
        at play.core.server.PekkoHttpServer.sslContext$lzyINIT1(PekkoHttpServer.scala:245)
        at play.core.server.PekkoHttpServer.sslContext(PekkoHttpServer.scala:245)
        at play.core.server.PekkoHttpServer.Http1Encrypted$lzyINIT1$$anonfun$2(PekkoHttpServer.scala:607)
        at scala.Option.map(Option.scala:242)
        at play.core.server.PekkoHttpServer.Http1Encrypted$lzyINIT1(PekkoHttpServer.scala:608)
        at play.core.server.PekkoHttpServer.Http1Encrypted(PekkoHttpServer.scala:597)
        at play.core.server.PekkoHttpServer.<init>(PekkoHttpServer.scala:641)
        at play.core.server.PekkoHttpServerProvider.createServer(PekkoHttpServer.scala:742)
        at play.core.server.PekkoHttpServerProvider.createServer(PekkoHttpServer.scala:741)
        at play.core.server.ServerProvider.createServer(ServerProvider.scala:30)
        at play.core.server.ServerProvider.createServer$(ServerProvider.scala:21)
        at play.core.server.PekkoHttpServerProvider.createServer(PekkoHttpServer.scala:740)
        at play.core.server.ProdServerStart$.start(ProdServerStart.scala:58)
        at play.core.server.ProdServerStart$.main(ProdServerStart.scala:28)
        at play.core.server.ProdServerStart.main(ProdServerStart.scala)
Caused by: java.lang.IllegalAccessError: class com.typesafe.sslconfig.ssl.FakeKeyStore (in unnamed module @0xfdc81>
        at com.typesafe.sslconfig.ssl.FakeKeyStore.certificateTooWeak(FakeKeyStore.scala:169)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.shouldGenerate$$anonfun$2$$anonfun$1(FakeKeyStore.scala:152)
        at scala.Option.exists(Option.scala:406)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.shouldGenerate$$anonfun$1(FakeKeyStore.scala:152)
        at scala.collection.IterableOnceOps.exists(IterableOnce.scala:604)
        at scala.collection.IterableOnceOps.exists$(IterableOnce.scala:601)
        at scala.collection.AbstractIterator.exists(Iterator.scala:1300)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.shouldGenerate(FakeKeyStore.scala:152)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.createKeyStore(FakeKeyStore.scala:179)
        at com.typesafe.sslconfig.ssl.FakeKeyStore.keyManagerFactory(FakeKeyStore.scala:218)
        at play.core.server.ssl.DefaultSSLEngineProvider.createSSLContext(DefaultSSLEngineProvider.scala:67)
        at play.core.server.ssl.DefaultSSLEngineProvider.<init>(DefaultSSLEngineProvider.scala:32)
        ... 22 more
</code>
</details>

### Play Version

3.0.2

### API

Scala 3.3.1

### Operating System

22.04.1-Ubuntu

### JDK

```
openjdk version "17.0.10" 2024-01-16
OpenJDK Runtime Environment (build 17.0.10+7-Ubuntu-122.04.1)
OpenJDK 64-Bit Server VM (build 17.0.10+7-Ubuntu-122.04.1, mixed mode, sharing)
```

### Expected Behavior

1. Server should start with autogenerated ssl cert.

### Actual Behavior

Server isn't started. (Stack trace see above.)

### Reproducible Test Case

Set up a empty Play project in InteliJ and run

`sbt -J"-Dhttp.port=disabled" -J"-Dhttps.port=9443" run`

--- a/documentation/manual/releases/release29/migration29/Migration29.md
+++ b/documentation/manual/releases/release29/migration29/Migration29.md
@@ -326,6 +326,8 @@ play.core.server.SelfSignedSSLEngineProvider.sslContext(SelfSigned.scala:36)
 ...
 ```
 
+#### Self-Signed Certificates in tests
+
 To circumvent this exception within a standard Play application setup, the [[binding of the HTTPS port for tests has been disabled|Migration29#Test-servers-no-longer-bind-an-HTTPS-port-by-default]] as a default configuration. It is noteworthy that the unbinding of the HTTPS port in tests is a reversion to an older behavior, as HTTPS port binding was unintentionally activated by us after being disabled in earlier Play versions.
 
 Should you desire to conduct tests against the HTTPS port while using a Play-generated self-signed certificate, a workaround is required for Java 17. It involves adding an `--add-export` flag to the Java 17 command line. Moreover, it is crucial to fork a new Java Virtual Machine (JVM) instance during test execution:
@@ -340,6 +342,50 @@ Alternatively, you can set the `JAVA_TOOL_OPTIONS` environment variable:
 ```sh
 export JAVA_TOOL_OPTIONS="$JAVA_TOOL_OPTIONS --add-exports=java.base/sun.security.x509=ALL-UNNAMED";
 ```
+
+#### Self-Signed Certificates in DEV mode
+
+If you, for whatever reason, serve your application [[in DEV mode via HTTPS|ConfigFile#Using-with-the-run-command]] using a Play-generated self-signed certificate, you also have to add the `--add-export` flag to the `sbt` command (which eventually starts a Java process in which your application runs):
+
+```sh
+# For example, to enter the interactive sbt shell:
+sbt -J"--add-exports=java.base/sun.security.x509=ALL-UNNAMED"
+
+# Or, mixed with other arguments and directly running the application:
+sbt -J"-Dhttp.port=disabled" -J"-Dhttps.port=9443" -J"--add-exports=java.base/sun.security.x509=ALL-UNNAMED" run
+```
+
+Instead of passing the flag on the command line, you can also put it into either a `.jvmopts` file within the root directory of your project
+
+```
+--add-exports=java.base/sun.security.x509=ALL-UNNAMED
+```
+
+or inside a `.sbtopts` file, again in the root of your project (be aware of the `-J` prefix):
+
+```
+-J--add-exports=java.base/sun.security.x509=ALL-UNNAMED
+```
+
+As a last option, you could, again, also set the `JAVA_TOOL_OPTIONS` environment variable.
+
+#### Self-Signed Certificates in production
+
+It's not recommended to serve websites via self-signed certificates in production. However, if you do so, e.g., to test [[the packaged application|Deploying]] locally, you also need to add the `--add-export` flag to the script that starts the Java process in which your application runs:
+
+```sh
+<my-project-script> -J--add-exports=java.base/sun.security.x509=ALL-UNNAMED #-Dhttp.port=disabled -Dhttps.port=9443
+```
+
+Instead of passing the flag via the command line, you can set it once in `build.sbt` and it will be appended to the start script automatically:
+
+```sbt
+Universal / javaOptions += "-J--add-exports=java.base/sun.security.x509=ALL-UNNAMED"
+```
+
+Alternatively, you can make use of the `JAVA_TOOL_OPTIONS` environment variable, as described above.
+
+#### Self-Signed Certificates in Java 21
 
 Unfortunately, the aforementioned workaround is not sufficient for Java 21. In this case, you may encounter the following exception:
 

--- a/documentation/manual/working/commonGuide/production/ConfiguringHttps.md
+++ b/documentation/manual/working/commonGuide/production/ConfiguringHttps.md
@@ -18,6 +18,8 @@ HTTPS configuration can either be supplied using system properties or in `applic
 
 By default, Play will generate itself a self-signed certificate, however typically this will not be suitable for serving a website.  Play uses Java key stores to configure SSL certificates and keys.
 
+> Please be aware that binding an HTTPS port with a self-signed certificate in Java 17 and Java 21 may lead to issues. For more details on this matter, refer to [["Generation of Self-Signed Certificates Fails in Java 17 and Java 21"|Migration29#Generation-of-Self-Signed-Certificates-Fails-in-Java-17-and-Java-21]] in the Play 2.9 Migration Guide.
+
 Signing authorities often provide instructions on how to create a Java keystore (typically with reference to Tomcat configuration).  The official Oracle documentation on how to generate keystores using the JDK keytool utility can be found [here](https://docs.oracle.com/en/java/javase/17/docs/specs/man/keytool.html).  There is also an example in the [Generating X.509 Certificates](https://lightbend.github.io/ssl-config/CertificateGeneration.html) section.
 
 Having created your keystore, the following configuration properties can be used to configure Play to use it:

